WIP - Refactor Task usage to Instance

unterstein opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mesosphere/marathon/4275)
<!-- Reviewable:end -->

jasongilanfarr commented on src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala:
```diff
@@ -391,8 +392,8 @@ object MarathonSchedulerActor {
 
-  case class KillTasks(appId: PathId, tasks: Iterable[Task]) extends Command {
-    def answer: Event = TasksKilled(appId, tasks.map(_.taskId))
+  case class KillTasks(appId: PathId, tasks: Iterable[Instance]) extends Command {
```
Let's take the opportunity to use immutable.Seq over Iterable.

jasongilanfarr commented on src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala:
```diff
@@ -401,7 +402,7 @@ object MarathonSchedulerActor {
   case object TasksReconciled extends Event
   case class DeploymentStarted(plan: DeploymentPlan) extends Event
-  case class TasksKilled(appId: PathId, taskIds: Iterable[Task.Id]) extends Event
+  case class TasksKilled(appId: PathId, taskIds: Iterable[Instance.Id]) extends Event
```
Ditto

jasongilanfarr commented on src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala:
```diff
@@ -150,7 +150,7 @@ class MarathonSchedulerService @Inject() (
     appId: PathId,
-    tasks: Iterable[Task]): Iterable[Task] = {
+    tasks: Iterable[Instance]): Iterable[Instance] = {
     schedulerActor ! KillTasks(appId, tasks)
```
Immutable Seq

jasongilanfarr commented on src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala:
```diff
@@ -587,25 +587,16 @@ class SchedulerActions(
-      }
+  def sortByStateAndTime(a: Instance, b: Instance): Boolean = {
+    runningOrStaged(b.state.status) compareTo runningOrStaged(a.state.status) match {
+      case 0 => (a.state.since compareTo b.state.since) > 0
```
Please use dot notation

unterstein replied:
yeah

jdef commented:
Review status: 0 of 131 files reviewed at latest revision, 25 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala, line 120 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_fzY2x9X_UwxLpCD5:-KQ_fzY3JCqU0QSaFJTX:bxev0lt) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala#L120)):*
> ```Scala
>     enumFormat(mesos.TaskState.valueOf, str => s"$str is not a valid TaskState type")
> 
>   implicit lazy val TaskWrites: Writes[Instance] = Writes {
> ```

s/TaskWrites/InstanceWrites/

---

*[src/main/scala/mesosphere/marathon/core/health/HealthResult.scala, line 7 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_gkJ4MbsV3Q5FOnhm:-KQ_gkJ4MbsV3Q5FOnhn:b-qhbirz) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/HealthResult.scala#L7)):*
> ```Scala
> 
> sealed trait HealthResult {
>   def taskId: Instance.Id
> ```

s/taskId/instanceId/

---

*[src/main/scala/mesosphere/marathon/core/health/HealthResult.scala, line 13 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_gm6-SzRbSo7nhBpt:-KQ_gm6-SzRbSo7nhBpu:b-7ropln) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/HealthResult.scala#L13)):*
> ```Scala
> 
> case class Healthy(
>   taskId: Instance.Id,
> ```

ditto

---

*[src/main/scala/mesosphere/marathon/core/health/HealthResult.scala, line 18 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_gnJ6nARYtD-HB8Um:-KQ_gnJ6nARYtD-HB8Un:b-7ropln) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/HealthResult.scala#L18)):*
> ```Scala
> 
> case class Unhealthy(
>   taskId: Instance.Id,
> ```

ditto

---

*[src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala, line 26 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_gssGm40DdXzuwZNI:-KQ_gssGm40DdXzuwZNJ:b-rpd54f) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala#L26)):*
> ```Scala
> 
>   var nextScheduledCheck: Option[Cancellable] = None
>   var taskHealth = Map[Instance.Id, Health]()
> ```

s/taskHealth/instanceHealth/

---

*[src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala, line 94 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_h2Kdt7QllAgFQXZF:-KQ_h2Kdt7QllAgFQXZG:bj8fwn6) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala#L94)):*
> ```Scala
>             log.debug("Dispatching health check job for {}", task.id)
>             val worker: ActorRef = context.actorOf(workerProps)
>             worker ! HealthCheckJob(app, task.asInstanceOf[Task], launched, healthCheck)
> ```

?asInstanceOf needed?

---

*[src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala, line 10 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_i_rRAqmu1hU3B1VJ:-KQ_i_rSdW0H8AJtYoBN:b-9p4d52) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala#L10)):*
> ```Scala
> 
> /** Infers which TaskOps to create for given run spec and offers. */
> trait TaskOpFactory {
> ```

InstanceOpFactory? that "Infers which InstanceOp's..." ?

---

*[src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala, line 25 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_iTsFjT4MTp7fhCps:-KQ_iTsFjT4MTp7fhCpt:bgoflel) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala#L25)):*
> ```Scala
>     * @param additionalLaunches the number of additional launches that has been requested
>     */
>   case class Request(runSpec: RunSpec, offer: Mesos.Offer, taskMap: Map[Instance.Id, Instance],
> ```

instanceMap?

---

*[src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala, line 28 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_iVOM8AKHsSpJ0rJ_:-KQ_iVOM8AKHsSpJ0rJa:b-8mp516) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala#L28)):*
> ```Scala
>       additionalLaunches: Int) {
>     def frameworkId: FrameworkId = FrameworkId("").mergeFromProto(offer.getFrameworkId)
>     def tasks: Iterable[Instance] = taskMap.values
> ```

instances?

---

*[src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala, line 36 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_iWthlBJ2qlGQgSDw:-KQ_iWthlBJ2qlGQgSDx:b-8mp516) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala#L36)):*
> ```Scala
> 
>   object Request {
>     def apply(runSpec: RunSpec, offer: Mesos.Offer, tasks: Iterable[Instance], additionalLaunches: Int): Request = {
> ```

instances?

---

*[src/main/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImpl.scala, line 5 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_j0DdgW8oWMwAU0PF:-KQ_j0De8veg_ZAzX2o8:b-qvbnvg) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImpl.scala#L5)):*
> ```Scala
> import akka.pattern.AskTimeoutException
> import mesosphere.marathon.core.base.Clock
> import mesosphere.marathon.core.launcher.{ OfferProcessor, OfferProcessorConfig, TaskLauncher, TaskOp }
> ```

InstanceOp?

---

*[src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLabels.scala, line 8 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_jJZDR7Ohzl_Ut7ZR:-KQ_jJZDR7Ohzl_Ut7ZS:b-hevw1m) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLabels.scala#L8)):*
> ```Scala
> import org.apache.mesos.{ Protos => MesosProtos }
> 
> object TaskLabels {
> ```

InstanceLabels?

---

*[src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLabels.scala, line 25 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_jZuyjvhdEzSt82Kk:-KQ_jZuzPFL6v4ZHNPYG:b-ts9auk) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLabels.scala#L25)):*
> ```Scala
>   }
> 
>   def labelsForTask(frameworkId: FrameworkId, task: Task): ReservationLabels =
> ```

instance: Instance ?

---

*[src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala, line 89 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_kp4E1pAVPfjGXhb4:-KQ_kp4E1pAVPfjGXhb5:b-swolek) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala#L89)):*
> ```Scala
>   // scalastyle:on parameter.number
> 
>   private[this] var inFlightTaskOperations = Map.empty[Instance.Id, Cancellable]
> ```

inFlightInstanceOperations?

---

*[src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala, line 95 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_krExmgS9VkvzVjSc:-KQ_krExmgS9VkvzVjSd:bke66n0) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala#L95)):*
> ```Scala
> 
>   /** tasks that are in flight and those in the tracker */
>   private[this] var tasksMap: Map[Instance.Id, Instance] = _
> ```

instancesMap?

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala, line 39 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_lPqarzYk1qbc3GOj:-KQ_lPqarzYk1qbc3GOk:bgks1oq) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala#L39)):*
> ```Scala
>   }
> 
>   def reserve(frameworkId: FrameworkId, taskId: Instance.Id, resources: Iterable[Mesos.Resource]): //
> ```

instanceId?

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala, line 66 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_lXYhkegxKEugiuHg:-KQ_lXYhkegxKEugiuHh:bgks1oq) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala#L66)):*
> ```Scala
>   def createVolumes(
>     frameworkId: FrameworkId,
>     taskId: Instance.Id,
> ```

instanceId?

---

*[src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala, line 42 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_lr4tjFPUHDuEi1Z6:-KQ_lr4tjFPUHDuEi1Z7:b-vomwi0) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala#L42)):*
> ```Scala
>     val frameworkId = FrameworkId("").mergeFromProto(offer.getFrameworkId)
> 
>     val resourcesByTaskId: Map[Instance.Id, Iterable[Resource]] = {
> ```

resourcesByInstanceId?

---

*[src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala, line 52 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_ltLC3w-F11Bl06ch:-KQ_ltLC3w-F11Bl06ci:b55jy07) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala#L52)):*
> ```Scala
>   }
> 
>   private[this] def processResourcesByTaskId(
> ```

processResourcesByInstanceId? with a TODO for pods support?

---

*[src/main/scala/mesosphere/marathon/core/task/Task.scala, line 350 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_n2VWeg4hgZEWhBvg:-KQ_n2VWeg4hgZEWhBvh:b-4lmuud) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/task/Task.scala#L350)):*
> ```Scala
> 
>   def tasksById(tasks: Iterable[Instance]): Map[Instance.Id, Instance] =
>     tasks.iterator.map(task => task.id -> task).toMap
> ```

should this filter by type Task? or is this deep enough that only Task's will appear in the Iterable anyway?

---

*[src/main/scala/mesosphere/marathon/core/task/tracker/TaskTracker.scala, line 26 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_oq4Tu8j0ducPWlJ1:-KQ_oq4UCz3_2iqg0Ilz:b-obk16l) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/task/tracker/TaskTracker.scala#L26)):*
> ```Scala
>   def appTasks(appId: PathId)(implicit ec: ExecutionContext): Future[Iterable[Instance]]
> 
>   def task(taskId: Instance.Id): Future[Option[Instance]]
> ```

the impl expects this to possibly return Pod things as well. Should TaskTracker be refactored into InstanceTracker and live elsewhere?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

unterstein commented:
Review status: 0 of 131 files reviewed at latest revision, 25 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala, line 120 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_fzY2x9X_UwxLpCD5:-KQaAsyPHdUlOv3ZMeL1:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala#L120)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> s/TaskWrites/InstanceWrites/
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/health/HealthResult.scala, line 7 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_gkJ4MbsV3Q5FOnhm:-KQaAyQHXip5EzwYogjs:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/HealthResult.scala#L7)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> s/taskId/instanceId/
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/health/HealthResult.scala, line 13 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_gm6-SzRbSo7nhBpt:-KQaB67-N4etvI9MwGRG:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/HealthResult.scala#L13)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> ditto
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/health/HealthResult.scala, line 18 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_gnJ6nARYtD-HB8Um:-KQaB6PpQY2PPH3lOdWu:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/HealthResult.scala#L18)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> ditto
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala, line 26 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_gssGm40DdXzuwZNI:-KQaBFrD53vBjnNpU_GF:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala#L26)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> s/taskHealth/instanceHealth/
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala, line 94 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_h2Kdt7QllAgFQXZF:-KQaBTyqJMh7HvfbC7Vd:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala#L94)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> ?asInstanceOf needed?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala, line 10 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_i_rRAqmu1hU3B1VJ:-KQbACm7SqDDUInyXQKE:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala#L10)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> InstanceOpFactory? that "Infers which InstanceOp's..." ?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala, line 25 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_iTsFjT4MTp7fhCps:-KQbA_7gw9CtHYfE1PDT:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala#L25)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> instanceMap?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala, line 28 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_iVOM8AKHsSpJ0rJ_:-KQbAZqt_EyoIa2mu5rv:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala#L28)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> instances?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala, line 36 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_iWthlBJ2qlGQgSDw:-KQbAaHO4hFbQiA7-0O7:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/TaskOpFactory.scala#L36)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> instances?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImpl.scala, line 5 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_j0DdgW8oWMwAU0PF:-KQbAcfdKTQn0crSIrJd:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImpl.scala#L5)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> InstanceOp?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala, line 89 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_kp4E1pAVPfjGXhb4:-KQbAskzJVt2HJ-17o1X:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala#L89)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> inFlightInstanceOperations?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala, line 95 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_krExmgS9VkvzVjSc:-KQbBPU2cPcrFhf28WKn:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala#L95)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> instancesMap?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala, line 39 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_lPqarzYk1qbc3GOj:-KQbBVINUJm5dtUMevTp:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala#L39)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> instanceId?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala, line 66 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_lXYhkegxKEugiuHg:-KQbBYIgf4QoM-u0K5rT:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala#L66)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> instanceId?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala, line 42 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_lr4tjFPUHDuEi1Z6:-KQbBeG2J5dNAF_IUnWE:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala#L42)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> resourcesByInstanceId?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala, line 52 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_ltLC3w-F11Bl06ch:-KQbC2a26M7QAyQxCdBq:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala#L52)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> processResourcesByInstanceId? with a TODO for pods support?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/task/Task.scala, line 350 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_n2VWeg4hgZEWhBvg:-KQbCLVs-bTX1bS6WlSj:b-jaas7w) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/task/Task.scala#L350)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> should this filter by type Task? or is this deep enough that only Task's will appear in the Iterable anyway?
</details>
Should be moved to Instance IMO. What do you think? It`s a helper to transform a list of Instances to a Map(id->Instance)

---

*[src/main/scala/mesosphere/marathon/core/task/tracker/TaskTracker.scala, line 26 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_oq4Tu8j0ducPWlJ1:-KQbCY2s3lzshYb1KYW2:btbbx5) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/task/tracker/TaskTracker.scala#L26)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> the impl expects this to possibly return Pod things as well. Should TaskTracker be refactored into InstanceTracker and live elsewhere?
</details>
Renamed to InstanceTracker -> wow, was a huge change!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

jdef commented:
Reviewed 131 of 148 files at r1, 1 of 1 files at r2.
Review status: 65 of 164 files reviewed at latest revision, 28 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryHelper.scala, line 10 \[r3\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQcr2I0Jl07QplJEoLZ:-KQcr2I1r4uEPl18auY7:b-xuozr2) ([raw file](https://github.com/mesosphere/marathon/blob/fc8ff04ff29712b5167cd7eb8485e787439b197c/src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryHelper.scala#L10)):*
> ```Scala
> import org.apache.mesos.{ Protos => Mesos }
> 
> class TaskOpFactoryHelper(
> ```

InstanceOpFactoryHelper?

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/TaskOpSourceDelegate.scala, line 8 \[r3\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQcrryZyNeQk18bD0ak:-KQcrryZyNeQk18bD0al:b-peicdj) ([raw file](https://github.com/mesosphere/marathon/blob/fc8ff04ff29712b5167cd7eb8485e787439b197c/src/main/scala/mesosphere/marathon/core/matcher/base/util/TaskOpSourceDelegate.scala#L8)):*
> ```Scala
> import mesosphere.marathon.core.matcher.base.util.TaskOpSourceDelegate.{ TaskOpAccepted, TaskOpRejected }
> 
> private class TaskOpSourceDelegate(actorRef: ActorRef) extends TaskOpSource {
> ```

s/TaskOp/InstanceOp/ everywhere in here?

---

*[src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala, line 71 \[r3\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQcsML6py2rkCkJljxt:-KQcsML7zMhH7DYKDcn2:b-463iwl) ([raw file](https://github.com/mesosphere/marathon/blob/fc8ff04ff29712b5167cd7eb8485e787439b197c/src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala#L71)):*
> ```Scala
>                   resources = spuriousResources.to[Seq]
>                 )
>               log.warn("removing spurious resources and volumes of {} because the app does no longer exist", instanceId)
> ```

s/app/instance/

---

*[src/main/scala/mesosphere/marathon/core/task/Task.scala, line 350 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_n2VWeg4hgZEWhBvg:-KQcsUv56PE0Eu3lXr5j:b-qlis4a) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/task/Task.scala#L350)):*
<details><summary><i>Previously, unterstein (Johannes Unterstein) wrote…</i></summary>
> Should be moved to Instance IMO. What do you think? It`s a helper to transform a list of Instances to a Map(id->Instance)
</details>
moving to Instance sounds good

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

jdef commented:
yep, tons of changes!

unterstein commented:
Review status: 65 of 164 files reviewed at latest revision, 28 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryHelper.scala, line 10 \[r3\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQcr2I0Jl07QplJEoLZ:-KQfDcb70fN7Ry3xOSkZ:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/fc8ff04ff29712b5167cd7eb8485e787439b197c/src/main/scala/mesosphere/marathon/core/launcher/impl/TaskOpFactoryHelper.scala#L10)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> InstanceOpFactoryHelper?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/TaskOpSourceDelegate.scala, line 8 \[r3\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQcrryZyNeQk18bD0ak:-KQfDjNDrdaBEMIa9Yct:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/fc8ff04ff29712b5167cd7eb8485e787439b197c/src/main/scala/mesosphere/marathon/core/matcher/base/util/TaskOpSourceDelegate.scala#L8)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> s/TaskOp/InstanceOp/ everywhere in here?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/task/Task.scala, line 350 \[r1\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQ_n2VWeg4hgZEWhBvg:-KQfE6wuyfkKwqj1bgyx:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/212e660f15069223ab4ce67613fa64063e9568d3/src/main/scala/mesosphere/marathon/core/task/Task.scala#L350)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> moving to Instance sounds good
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

jdef commented:
Reviewed 33 of 35 files at r3, 100 of 100 files at r4.
Review status: 148 of 164 files reviewed at latest revision, 32 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 14 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfIAdVotvR0oPLlRDi:-KQfIAdVotvR0oPLlRDj:be8jqvd) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L14)):*
> ```Scala
> 
> object InstanceOpSourceDelegate {
>   def apply(actorRef: ActorRef): TaskOpSource = new InstanceOpSourceDelegate(actorRef)
> ```

InstanceOpSource?

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 16 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI1DqXpLGvG1vmQ7T:-KQfI1DrlVCB56vClwHN:bgo66ux) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L16)):*
> ```Scala
>   def apply(actorRef: ActorRef): TaskOpSource = new InstanceOpSourceDelegate(actorRef)
> 
>   sealed trait TaskOpNotification {
> ```

InstanceOpNotification?

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 22 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI4riv55MI_5UHLiT:-KQfI4riv55MI_5UHLiU:b-xp9ymb) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L22)):*
> ```Scala
>     def unapply(notification: TaskOpNotification): Option[InstanceOp] = Some(notification.taskOp)
>   }
>   case class TaskOpAccepted(taskOp: InstanceOp) extends TaskOpNotification
> ```

InstanceOpAccepted?

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 23 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI6Mala-MDe0i8WSD:-KQfI6Mala-MDe0i8WSE:b-qiqg6y) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L23)):*
> ```Scala
>   }
>   case class TaskOpAccepted(taskOp: InstanceOp) extends TaskOpNotification
>   case class TaskOpRejected(taskOp: InstanceOp, reason: String) extends TaskOpNotification
> ```

InstanceOpRejected?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

unterstein commented:
Review status: 148 of 164 files reviewed at latest revision, 32 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala, line 71 \[r3\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQcsML6py2rkCkJljxt:-KQfIniTwqFSBAFnTuHX:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/fc8ff04ff29712b5167cd7eb8485e787439b197c/src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala#L71)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> s/app/instance/
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

unterstein commented:
Review status: 147 of 164 files reviewed at latest revision, 32 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 14 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfIAdVotvR0oPLlRDi:-KQfOnA2hnk3MwTsTbw0:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L14)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> InstanceOpSource?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 16 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI1DqXpLGvG1vmQ7T:-KQfOlVa11lpWG0Vegcd:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L16)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> InstanceOpNotification?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 22 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI4riv55MI_5UHLiT:-KQfOm6oCWzFbtVPIjMx:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L22)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> InstanceOpAccepted?
</details>
Done.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 23 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI6Mala-MDe0i8WSD:-KQfOmlEVW9Xsr0lj7Q2:b-896fix) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L23)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> InstanceOpRejected?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

jdef commented:
Reviewed 3 of 35 files at r3, 18 of 18 files at r5.
Review status: 157 of 164 files reviewed at latest revision, 32 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 16 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI1DqXpLGvG1vmQ7T:-KQfWSJC93lGFfP9CSJd:beshmrr) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L16)):*
<details><summary><i>Previously, unterstein (Johannes Unterstein) wrote…</i></summary>
> Done.
</details>
Any particular reason why  Op was dropped from the name?

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 22 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI4riv55MI_5UHLiT:-KQfWK2k89kOWBRBzTF9:beshmrr) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L22)):*
<details><summary><i>Previously, unterstein (Johannes Unterstein) wrote…</i></summary>
> Done.
</details>
Any particular reason why  Op was dropped from the name?

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 23 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI6Mala-MDe0i8WSD:-KQfWOLVq3OKYKA-zOUg:beshmrr) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L23)):*
<details><summary><i>Previously, unterstein (Johannes Unterstein) wrote…</i></summary>
> Done.
</details>
Any particular reason why  Op was dropped from the name?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

unterstein commented:
Review status: 146 of 164 files reviewed at latest revision, 32 unresolved discussions, some commit checks failed.

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 16 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI1DqXpLGvG1vmQ7T:-KQfYAKNkppGTrRueDSi:bxziahr) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L16)):*
> InstanceOpNotification
Oooops! No, thanks!

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 22 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI4riv55MI_5UHLiT:-KQfZRf19MaMmNkfeRpc:bu1b5ya) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L22)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> Any particular reason why  Op was dropped from the name?
</details>
Oooops, reverted!

---

*[src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala, line 23 \[r5\]](https://reviewable.io:443/reviews/mesosphere/marathon/4275#-KQfI6Mala-MDe0i8WSD:-KQfZNCpsW3oneEYNHXb:bu1b5ya) ([raw file](https://github.com/mesosphere/marathon/blob/e614d51992dfc1308d426283c5c3350bdbbcc1c2/src/main/scala/mesosphere/marathon/core/matcher/base/util/InstanceOpSourceDelegate.scala#L23)):*
<details><summary><i>Previously, jdef (James DeFelice) wrote…</i></summary>
> Any particular reason why  Op was dropped from the name?
</details>
Oooops, reverted!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mesosphere/marathon/4275)*
<!-- Sent from Reviewable.io -->

jdef commented:
lgtm once build passes

jdef commented:
:clap:
